adr/0017-decouple-transformation-library-from-zarf.md

```diff
@@ -0,0 +1,22 @@
+# NUMBER. TITLE
+
+Date: 2023-05-28
+
+## Status
+
+TBD
```

Comment:
Pending until your gRPC Pr is merged

Reply:
Addressed in https://github.com/defenseunicorns/zarf/pull/1900